template/src/api/index.js

```diff
@@ -87,13 +89,15 @@ export default function indexEntrypointFile({asyncapi, params}) {
     }
     `;
 
-    const handlers = Object.entries(asyncapi.channels()).map(([channelName, channel]) => {
+    const handlers = asyncapi.channels().all().map(channel => {
+        const channelName = channel.address();
         let handler = '';
-        if (channel.hasPublish()) {
-            handler += `${convertOpertionIdToMiddlewareFn(channel.publish().id())} : require('./handlers/${convertToFilename(channelName)}').${convertOpertionIdToMiddlewareFn(channel.publish().id())},`;
-        }
-        if (channel.hasSubscribe()) {
-            handler += `${convertOpertionIdToMiddlewareFn(channel.subscribe().id())} : require('./handlers/${convertToFilename(channelName)}').${convertOpertionIdToMiddlewareFn(channel.subscribe().id())},`;
+        for (let operation of channel.operations()) {
+            const operationId = operation.id();
+            if (operation.isReceive())
+            handler += `${convertOpertionIdToMiddlewareFn(operationId)} : require('./handlers/${convertToFilename(channelName)}').${convertOpertionIdToMiddlewareFn(operationId)},`;
+            if (operation.isSend())
+            handler += `${convertOpertionIdToMiddlewareFn(operationId)} : require('./handlers/${convertToFilename(channelName)}').${convertOpertionIdToMiddlewareFn(operationId)},`;
```

Comment:
what are the differences here so that it is conditioned with `isReceive` and `isSend`

you no longer need `channel.publish().id()` so there is no difference, right?